### PR TITLE
Fix/condo/sberdoma 721/resident api fixes

### PR DESCRIPTION
**Pull request by AntonAL**
SBERDOMA-721 virtual field `propertyForResident` for `Resident` schema
SBERDOMA-721 virtual field `organizationForResident` for `Resident` schema
SBERDOMA-953 test on soft-delete for Resident
SBERDOMA-953 forbid to change `address`, `addressMeta`, `unitName` or `property` of Resident
SBERDOMA-947 allow to create Resident with same address and unitName for different users

Sorry for several tasks, because they needs to be delivered at once now

### CODE REVIEW

File: apps/condo/domains/resident/schema/Resident.js [hunk 1]
```diff
@@ -40,6 +42,22 @@ const Resident = new GQLListSchema('Resident', {
                 update: false,
             },
         },
+        // The reason for this field is to avoid adding check for resident user into global Organization read access.
+        // This field have specific use case for mobile client.
+        organizationForResident: {
+            schemaDoc: 'Organization data, that is returned for current resident in mobile client',
+            type: Virtual,
+            graphQLReturnType: 'JSON',
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/Resident.js:50]:
```suggestion
            extendGraphQLTypes: [`type ResidentOrganization { id: ID!, name: String }`],
            graphQLReturnType: 'ResidentOrganization',
```

File: apps/condo/domains/resident/schema/Resident.js [hunk 2]
```diff
@@ -61,6 +79,23 @@ const Resident = new GQLListSchema('Resident', {
             },
         },
 
+        // The reason for this field is to avoid adding check for resident user into global Property read access.
+        // This field have specific use case for mobile client.
+        propertyForResident: {
+            schemaDoc: 'Property data, that is returned for current resident in mobile client',
+            type: Virtual,
+            graphQLReturnType: 'JSON',
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/Resident.js:87]:
```suggestion
            extendGraphQLTypes: [`type ResidentProperty { id: ID!, address: String, name: String }`],
            graphQLReturnType: 'ResidentProperty',
```

File: apps/condo/domains/resident/schema/Resident.js [hunk 3]
```diff
@@ -61,6 +79,23 @@ const Resident = new GQLListSchema('Resident', {
             },
         },
 
+        // The reason for this field is to avoid adding check for resident user into global Property read access.
+        // This field have specific use case for mobile client.
+        propertyForResident: {
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/Resident.js:84]: residentProperty?

Comment (AntonAL) [reply apps/condo/domains/resident/schema/Resident.js:84]: Better to use meaningful word first, it's easier to read GraphQL query fields:

```
propertyForResident { ... }
organizationForResident { ... }
```

Comment (AntonAL) [reply apps/condo/domains/resident/schema/Resident.js:84]: Ah, we have GraphQL-types `ResidentProperty`, — `residentProperty` is better in this case.

File: apps/condo/domains/resident/gql.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ const gql = require('graphql-tag')
 
 const COMMON_FIELDS = 'id dv sender v deletedAt newId createdBy { id name } updatedBy { id name } createdAt updatedAt'
 
-const RESIDENT_FIELDS = `{ user { id name } organization { id } property { id } billingAccount { id } address addressMeta unitName ${COMMON_FIELDS} }`
+const RESIDENT_FIELDS = `{ user { id name } organization { id } residentOrganization { id name } property { id } residentProperty { id name } billingAccount { id } address addressMeta unitName ${COMMON_FIELDS} }`
```
Comment (pahaz) [inline apps/condo/domains/resident/gql.js:13]: add property resident address

File: apps/condo/domains/resident/schema/Resident.js [hunk 5]
```diff
@@ -61,6 +80,24 @@ const Resident = new GQLListSchema('Resident', {
...
+        // This field have specific use case for mobile client.
+        residentProperty: {
+            schemaDoc: 'Property data, that is returned for current resident in mobile client',
+            type: Virtual,
+            extendGraphQLTypes: ['type ResidentProperty { id: ID!, name: String }'],
+            graphQLReturnType: 'ResidentProperty',
+            resolver: async (item) => {
+                if (item.property) {
+                    const property = await getById('Property', item.property)
+                    return pick(property, ['id', 'name'])
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/Resident.js:93]: add address

File: apps/condo/domains/resident/utils/testSchema/index.js [hunk 6]
```diff
@@ -53,13 +53,26 @@ async function updateTestResident (client, id, extraAttrs = {}) {
...
         dv: 1,
         sender,
-        unitName: faker.random.alphaNumeric(3),
         ...extraAttrs,
     }
     const obj = await Resident.update(client, id, attrs)
     return [obj, attrs]
 }
 
+async function softDeleteTestResident (client, id, extraAttrs = {}) {
```
Comment (pahaz) [inline apps/condo/domains/resident/utils/testSchema/index.js:62]: You can use update. We usually don’t create an until for that

File: apps/condo/domains/resident/gql.js (hunk 4 above)
Comment (AntonAL) [reply apps/condo/domains/resident/gql.js:13]: We already have `address` property, it's a scalar.

File: apps/condo/domains/resident/utils/testSchema/index.js (hunk 6 above)
Comment (AntonAL) [reply apps/condo/domains/resident/utils/testSchema/index.js:62]: I think, soft-delete is a specific case, that needs to pay extra attention, is't?
BTW, we already have `softDeleteTestOrganizationEmployee`.

File: apps/condo/domains/resident/gql.js (hunk 4 above)
Comment (AntonAL) [reply apps/condo/domains/resident/gql.js]: Added in `a6160203`

**Review by Dimitreee — APPROVED** [review]

File: apps/condo/domains/resident/utils/testSchema/index.js (hunk 6 above)
Comment (pahaz) [reply apps/condo/domains/resident/utils/testSchema/index.js:62]: it's better to add it to the core `generateServerUtils` ? to avoid code duplication?

Comment (AntonAL) [reply apps/condo/domains/resident/utils/testSchema/index.js:62]: Later, not in this task, please.
Created a separate task SBERDOMA-965 for that.

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/resident/utils/testSchema/index.js (hunk 6 above)
Comment (pahaz) [reply apps/condo/domains/resident/utils/testSchema/index.js:62]: yes! ship it

**Pull request merged**